pkgs/shells/fish/default.nix

```diff
@@ -59,6 +59,9 @@ stdenv.mkDerivation rec {
   '' + ''
     sed -i "s|/sbin /usr/sbin||" \
            "$out/share/fish/functions/__fish_complete_subcommand_root.fish"
+
+    # make fish pick up completions from nix profile
+    printf "set fish_complete_path (printf \$NIX_PROFILES | tr ' ' '\\\\n')\"/share/fish/vendor_completions.d\" \$fish_complete_path" >> $out/share/fish/config.fish
```

Comment:
But the way you're using it explodes even more badly if there's a `%` in the text.

Also the outer printf (that appends to $out/share/fish/config.fish) is being called with a constant string and makes the file not end in a newline.